redfish-core/lib/log_services.hpp

```diff
@@ -1138,6 +1330,23 @@ inline void createDump(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
             messages::internalError(asyncResp->res);
             return;
         }
+        if (oemDiagnosticDataType)
+        {
+            if (*oemDiagnosticDataType == "FaultData")
+            {
+                // Fault data dump
+                createDumpParams.emplace_back(
+                    "xyz.openbmc_project.Dump.Internal.Create.Type",
+                    "xyz.openbmc_project.Dump.Internal.Create.Type.FaultData");
+            }
+            else
+            {
+                BMCWEB_LOG_ERROR(
+                    "Wrong parameter value passed for 'OEMDiagnosticDataType'");
+                messages::internalError(asyncResp->res);
```

Comment:
This should not be an internal error, this should be a 400 (user error) and a warning log

Reply:
Ack